### DISCUSSION TITLE
How do we parameterize the tests in SeleniumBase

### DISCUSSION THREAD

**Discussion by Kumaravasanth** [seleniumbase/SeleniumBase#938] [Q&A] [answered]
Can we use the pytest fixtures to parameterize the tests created using SeleniumBase? Please share if there are any examples for the same.

**Comment by mdmintz** [comment, marked as answer]
There are two ways to parameterize SeleniumBase tests:
* With the ``parameterized`` library.
* With ``pytest.mark.parametrize``.

Here's an example of using ``parameterized``: https://github.com/seleniumbase/SeleniumBase/blob/master/examples/parameterized_test.py

Here's an example of using ``pytest.mark.parametrize``:
https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_pytest_parametrize.py

You'll notice that in order to use ``pytest.mark.parametrize``, you'll need to use SeleniumBase as a pytest fixture. That's one of the syntax formats of SeleniumBase. (See https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md for the full list of SeleniumBase syntax formats.)
